data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/s3partition/S3PartitionCreatorScheduler.java

```diff
@@ -25,13 +24,13 @@ public S3PartitionCreatorScheduler(final EnhancedSourceCoordinator sourceCoordin
     public void run() {
         while (!Thread.currentThread().isInterrupted()) {
             try {
-                final Optional<EnhancedSourcePartition> sourcePartition = sourceCoordinator.acquireAvailablePartition(S3Partition.PARTITION_TYPE);
+                final Optional<EnhancedSourcePartition> sourcePartition = sourceCoordinator.acquireAvailablePartition(S3FolderPartition.PARTITION_TYPE);
                 if (sourcePartition.isPresent()) {
-                    final S3Partition s3Partition = (S3Partition) sourcePartition.get();
-                    final List<String> s3Folders = createS3BucketPartitions(s3Partition);
-                    sourceCoordinator.completePartition(s3Partition);
-                    final S3PartitionStatus s3PartitionStatus = new S3PartitionStatus(s3Folders.size());
-                    sourceCoordinator.createPartition(new GlobalState(S3_FOLDER_PREFIX + s3Partition.getCollection(), s3PartitionStatus.toMap()));
+                    final S3FolderPartition s3FolderPartition = (S3FolderPartition) sourcePartition.get();
+                    final List<String> s3Folders = createS3BucketPartitions(s3FolderPartition);
+                    sourceCoordinator.completePartition(s3FolderPartition);
```

Comment:
I'm not sure what the point of this is with creating the bucket partitions and completing this folder partition

Reply:
This is making sure only one node picks up S3 folder partition creation job. More than one node shouldn't create them. In followup CR I can break out and end this thread.